### PR TITLE
Add feature for drawing rectangle with holding shift for selecting po…

### PR DESCRIPTION
**Pull request by LeonardKoch**
I created a quick feature for selecting points using a rectangle selection.
You can active it by holding shift and then clicking and dragging the rectangle across the map.
Not fully tested yet, but good to have a look already.
Let me know what you think.

### CODE REVIEW

File: src/PolygonDraw/Map.tsx [hunk 1]
```diff
@@ -183,19 +189,67 @@ export class BaseMap extends React.Component<Props, State> {
...
+            this.setState({
+                isRectangleSelectionDragActive: true,
+                rectangleSelectionStartPosition: coordinate,
+                rectangleSelectionEndPosition: coordinate
+            });
+        }
+    };
+
+    handleMouseUpOnMap = () => {
+        if (this.state.isRectangleSelectionDragActive) {
```
Comment (medhoover) [inline src/PolygonDraw/Map.tsx]: I'm not certain about this, but the condition could keep the rectangle visible if the user releases the shift key before releasing the mouse click.

File: src/PolygonDraw/Map.tsx [hunk 2]
```diff
@@ -60,9 +60,12 @@ export interface State {
     isMovedPointInBoundary: boolean;
     isShiftPressed: boolean;
     isMoveActive: boolean;
+    isRectangleSelectionDragActive: boolean;
+    rectangleSelectionStartPosition: Coordinate|null;
+    rectangleSelectionEndPosition: Coordinate|null;
```
[the remark this answers is not in the document]
Comment (LeonardKoch) [reply src/PolygonDraw/Map.tsx]: Mhm good idea.
What do you think about omitting the isRectangleDragActive property and just using the RectangleSelectionPosition not being null as the indicator for a drag being active?
Less code but slightly more implicit.

File: src/PolygonDraw/Map.tsx (hunk 1 above)
Comment (LeonardKoch) [reply src/PolygonDraw/Map.tsx]: Yeah I did that intentionally as an affordance for the user, so they don't have to hold down the shift key. I also didn't see any useful behaviour you could get out of stopping the drag if the shift key is released before the mouse is.

Comment (medhoover) [reply src/PolygonDraw/Map.tsx]: Sounds good. LGTM

**Pull request merged**